api/src/main/java/com/strategyobject/substrateclient/api/ApiBuilder.java

```diff
@@ -0,0 +1,50 @@
+package com.strategyobject.substrateclient.api;
+
+import com.google.inject.Injector;
+import com.strategyobject.substrateclient.common.types.AutoRegistry;
+import com.strategyobject.substrateclient.rpc.registries.RpcDecoderRegistry;
+import com.strategyobject.substrateclient.rpc.registries.RpcEncoderRegistry;
+import com.strategyobject.substrateclient.scale.registries.ScaleReaderRegistry;
+import com.strategyobject.substrateclient.scale.registries.ScaleWriterRegistry;
+import com.strategyobject.substrateclient.transport.ProviderInterface;
+import lombok.val;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+public class ApiBuilder {
+    private static final String ROOT_PREFIX = "com.strategyobject.substrateclient";
+
+    private final Injector injector;
+
+    public ApiBuilder(Injector injector) {
+        this.injector = injector;
+
+        autoRegister(
+                ScaleReaderRegistry.class,
+                ScaleWriterRegistry.class,
+                RpcDecoderRegistry.class,
+                RpcEncoderRegistry.class);
+    }
+
+    public <T> ApiBuilder configure(Class<T> clazz, Consumer<T> consumer) {
```

Comment:
this method is done mainly to work with registries in a generic way. It can surprise you, though, if you override dependencies. But in this case I expect you to understand what you are doing. What do you think?

Reply:
♫ Miley Cyrus' voice sounds:

Reply:
cool, I like it